### PR TITLE
Applies in development SunPy package template

### PR DESCRIPTION
**Pull request by CyclingNinja**
Applies in development SunPy package template.

The main goal of the template is to unify the SunPy sub-packages to the same standards e.g. linting and dependancies.
This is the first step in developing the overall template specific to SunPy.

### CODE REVIEW

**Comment by Cadair** [comment]
@CyclingNinja #7384 has now been merged so rebase away :laughing:

File: pyproject.toml [hunk 1]
```diff
@@ -5,6 +5,9 @@ requires = [
   "setuptools>=61.2",
   "setuptools_scm[toml]>=6.2",
   "wheel",
+  "extension-helpers",
+  "oldest-supported-numpy",
+  "cython"
```
Comment (Cadair) [inline pyproject.toml:10]:
```suggestion
```

We should update the template to drop `oldest-supported-numpy` (https://github.com/OpenAstronomy/packaging-guide/issues/46) and `extension-helpers` is a duplicate here, and we don't use `cython`. I wonder if we should remove cython from the template or not :thinking:  we use it for the example stuff, but you don't always *need* it if compiled extensions is true.

File: sunpy/_compiler.c
Comment (Cadair) [inline sunpy/_compiler.c:1]: Where did this file come from? It's not even in the template?

File: sunpy/version.py [hunk 2]
```diff
@@ -15,9 +15,3 @@
     del warnings
 
     version = '0.0.0'
-
-from packaging.version import parse as _parse
-
-_version = _parse(version)
-major, minor, bugfix = [*_version.release, 0][:3]
-release = not _version.is_devrelease
```
Comment (Cadair) [inline sunpy/version.py:23]: I assume we are using this, but I wonder if we should add it to the template or not, given it introduces a packaging dependency.

File: licenses/TEMPLATE_LICENSE.rst [hunk 3]
```diff
@@ -0,0 +1,31 @@
+This project is based upon the Astropy package template
```
Comment (Cadair) [inline licenses/TEMPLATE_LICENSE.rst:1]: We should do a find and replace in the open astronomy package template for "astropy" lol

File: licenses/README.rst [hunk 4]
```diff
@@ -1,7 +1,9 @@
...
+========
 
-This directory holds license and credit information for works sunpy is derived from or distributes.
+This directory holds license and credit information for the package,
+works the package is derived from, and/or datasets.
 
-The license file for the sunpy package itself is placed in the root directory of this repository as LICENSE.rst.
+Ensure that you pick a package licence which is in this folder and it matches
+the one mentioned in the top level README.rst file. If you are using the
+pre-rendered version of this template check for the word 'Other' in the README.
```
Comment (Cadair) [inline licenses/README.rst:9]: This being in the rendered output is a curious choice on behalf of the template.

File: licenses/LICENSE.rst
Comment (Cadair) [inline licenses/LICENSE.rst:1]: This file should be removed, I think we want to keep the pattern of having the package license in the root of the package directory (we need to decide if that's a thing we want to do in openastronomy or sunpy templates) as there are a lot of things which expect it to be there.

File: .gitignore [hunk 5]
```diff
@@ -24,10 +24,17 @@ parts/
 sdist/
 var/
 wheels/
+share/python-wheels/
 *.egg-info/
 .installed.cfg
 *.egg
 MANIFEST
+sunpy/_version.py
+# PyInstaller
```
Comment (nabobalis) [inline .gitignore:33]:
```suggestion
sunpy/_version.py

# PyInstaller
```

Does cruft account for the package name here?

File: sunpy/tests/__init__.py [hunk 6]
```diff
@@ -0,0 +1,4 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
```
Comment (nabobalis) [inline sunpy/tests/__init__.py:1]: Do we need this?

File: sunpy/version.py (hunk 2 above)
Comment (nabobalis) [reply sunpy/version.py:23]: I do not think we are?

**Comment by nabobalis** [comment]
Does https://github.com/sunpy/sunpy/pull/7494 supercede this PR?

**Comment by nabobalis** [comment]
> Does #7494 supercede this PR?

I am going to assume yes and close this.

**Pull request closed without merging**